### PR TITLE
fix: show tooltip on hover long text

### PR DESCRIPTION
**Pull request by ChenCodes**
Changes in this PR:
- name is a string and not React.Node
- show browser tooltip for both truncated and non-truncated text in DataListItem
Image from iOS (1)

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @ChenCodes has signed the CLA. Thanks for the pull request!

File: src/components/datalist-item/DatalistItem.js [hunk 1]
```diff
@@ -46,14 +47,14 @@ class DatalistItem extends React.Component<Props> {
     id: string;
 
     render() {
-        const { children, className, isActive, ...rest } = this.props;
+        const { children, className, isActive, title, ...rest } = this.props;
```
Comment (priyajeet) [inline src/components/datalist-item/DatalistItem.js]: title already part of rest btw which in turn is part of itemprops

File: src/features/unified-share-modal/ContactsField.js [hunk 2]
```diff
@@ -198,7 +200,12 @@ class ContactsField extends React.Component<Props, State> {
                 validator={validator}
             >
                 {contacts.map(({ email, text = null, id }) => (
-                    <ContactDatalistItem key={id} name={text} subtitle={email || groupLabel} />
+                    <ContactDatalistItem
+                        key={id}
+                        name={text}
+                        subtitle={email || groupLabel}
+                        title={email && email.length > MAX_NUM_CHAR_FOR_LISTITEM ? email : ''}
```
Comment (priyajeet) [inline src/features/unified-share-modal/ContactsField.js]: You want to show browser tooltip or our own `<Tooltip>`?

File: src/components/datalist-item/DatalistItem.js (hunk 1 above)
Comment (ChenCodes) [reply src/components/datalist-item/DatalistItem.js]: It looks like when I take out title from props, I'm getting an error saying that title is not defined:
Screen Shot 2019-06-03 at 10 36 07 AM

File: src/features/unified-share-modal/ContactsField.js (hunk 2 above)
Comment (ChenCodes) [reply src/features/unified-share-modal/ContactsField.js]: Yes, verified with design

File: src/components/datalist-item/DatalistItem.js [hunk 3]
```diff
@@ -46,14 +47,14 @@ class DatalistItem extends React.Component<Props> {
...
-        const { children, className, isActive, ...rest } = this.props;
+        const { children, className, isActive, name, ...rest } = this.props;
         const classes = classNames('datalist-item', { 'is-active': isActive }, className);
         const itemProps = omit(rest, ['closeDropdown', 'setActiveItemID']);
 
         // required aria props are added dynamically
         /* eslint-disable jsx-a11y/role-has-required-aria-props */
         return (
-            <li {...itemProps} className={classes} id={this.id} role="option">
+            <li {...itemProps} className={classes} id={this.id} role="option" title={name}>
```
Comment (priyajeet) [inline src/components/datalist-item/DatalistItem.js]: title cannot be a React.Node, only string

File: src/components/datalist-item/DatalistItem.js [hunk 4]
```diff
@@ -46,14 +47,14 @@ class DatalistItem extends React.Component<Props> {
     id: string;
 
     render() {
-        const { children, className, isActive, ...rest } = this.props;
+        const { children, className, isActive, name, ...rest } = this.props;
```
Comment (priyajeet) [inline src/components/datalist-item/DatalistItem.js]: Why not just leave it to use title?

File: src/components/datalist-item/DatalistItem.js (hunk 3 above)
Comment (ChenCodes) [reply src/components/datalist-item/DatalistItem.js]: Turns out the flowtype for name should be string and not React.Node. Will be using name prop instead of title prop

File: src/components/contact-datalist-item/ContactDatalistItem.js [hunk 5]
```diff
@@ -6,8 +6,9 @@ import DatalistItem from '../datalist-item';
 import './ContactDatalistItem.scss';
 
 type Props = {
-    name: React.Node,
+    name: string | null,
```
Comment (priyajeet) [inline src/components/contact-datalist-item/ContactDatalistItem.js]: ?string

File: src/components/contact-datalist-item/ContactDatalistItem.js [hunk 6]
```diff
@@ -6,8 +6,9 @@ import DatalistItem from '../datalist-item';
...
     subtitle?: React.Node,
+    title?: string | null,
```
Comment (priyajeet) [inline src/components/contact-datalist-item/ContactDatalistItem.js]: why is this needed?

Comment (ChenCodes) [reply src/components/contact-datalist-item/ContactDatalistItem.js]: Not needed. It'll be necessary in the child component but not in this file

File: src/components/contact-datalist-item/ContactDatalistItem.js (hunk 5 above)
Comment (ChenCodes) [reply src/components/contact-datalist-item/ContactDatalistItem.js]: Fixed

File: src/components/datalist-item/DatalistItem.js [hunk 7]
```diff
@@ -53,7 +54,7 @@ class DatalistItem extends React.Component<Props> {
         // required aria props are added dynamically
         /* eslint-disable jsx-a11y/role-has-required-aria-props */
         return (
-            <li {...itemProps} className={classes} id={this.id} role="option">
+            <li {...itemProps} className={classes} id={this.id} role="option" title={rest.title}>
```
Comment (priyajeet) [inline src/components/datalist-item/DatalistItem.js]: rest is already part of itemProps, it will automatically apply the title, don't need to specify is explicitly

File: src/components/datalist-item/DatalistItem.js [hunk 8]
```diff
@@ -15,6 +15,7 @@ type Props = {
     isActive?: boolean,
     /** Set by a parent datalist component to receive the updated active item ID */
     setActiveItemID?: Function,
+    title?: string | null,
```
Comment (priyajeet) [inline src/components/datalist-item/DatalistItem.js]: not needed
